smdebug/tensorflow/keras.py

```diff
@@ -1295,37 +1321,51 @@ def save_tape_logs(self, model_inputs=None, outputs=None):
             self._log_unsupported_tensors_in_non_eager_execution()
 
     def wrap_tape(self, tape):
-        """
-        Wrapping your GradientTape with this method enables finding gradient tensors and optimizer
-        variables.
+        # Don't modify the original tape object. If an error occurs when wrapping the tape functions, we simply return
+        # the original tape object.
+        wrapped_tape = deepcopy(tape)
```

Comment:
Size isn't an issue with the tape object, since nested GradientTape is a valid use case. Doing a deep copy is the same as creating another GradientTape object with the same attributes.

Basically, we're copying the original tape object so that it purposefully stays unmodified. If an error occurs in wrapping the copy, we simply return the original tape object. The scope is only wrapping the tape, not the actual tape functions themselves.